llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -3614,6 +3616,13 @@ class BoUpSLP {
       DenseMap<Value *, SmallPtrSet<const TreeEntry *, 4>>;
   ValueToGatherNodesMap ValueToGatherNodes;
 
+  /// A list of the loads, which can be vectorized using strided or masked
+  /// gather approach, but attempted to be represented as contiguous loads.
+  SetVector<unsigned> LoadEntriesToVectorize;
+
+  /// true if graph nodes transforming mode is on.
+  bool IsGraphTransformMode = false;
+
   /// The index of the first gathered load entry in the VectorizeTree.
   constexpr static int NoGatheredLoads = -1;
   int GatheredLoadsEntriesFirst = NoGatheredLoads;
```

Comment:
(off topic) - Could this be `std::optional<unsigned> GatheredLoadsEntriesFirst = None` ?

Reply:
Yes, if you prefer this style. Can do in a separate patch